src/comestible_inv.cpp

```diff
@@ -0,0 +1,780 @@
+        desc += colorize( string_format( "%c", status[0] ), c_cyan );
+        desc += colorize( string_format(
+                              _( ": this food is frozen and cannot be consumed. Press %s to attempt reheating it.\n" ), status ),
+                          c_light_gray );
+
+        status = ctxt.get_desc( "CRAFT_WITH" );
+        desc += colorize( string_format( "%c", status[0] ), c_cyan );
+        desc += colorize( string_format(
+                              _( ": this item cannot be eaten in its current form. Press %s to attempt cooking with it.\n" ),
+                              status ), c_light_gray );
+        desc += "\n";
+
+        desc += colorize( string_format( "%c", 'm' ), c_yellow );
+        desc += colorize( _( ": this food is mushy and has reduced joy.\n" ), c_light_gray );
+
+        desc += colorize( string_format( "%c", 's' ), c_yellow );
+        desc += colorize( _( ": this is stimulant (e.g. coffe, pop).\n" ), c_light_gray );
+
+        desc += colorize( string_format( "%c", 'd' ), c_yellow );
+        desc += colorize( _( ": this is depressant (e.g. alcohol).\n" ), c_light_gray );
+        desc += "\n";
+
+        desc += colorize( string_format( "%c", 'a' ), c_yellow );
+        desc += colorize( _( ": this item is mildly addictive.\n" ), c_light_gray );
+
+        desc += colorize( string_format( "%c", 'a' ), c_red );
+        desc += colorize( _( ": this item is strongly addictive.\n" ), c_light_gray );
+
+        desc += colorize( "!", c_light_red );
+        desc += colorize( string_format( _( ": eating this will be considered stealing." ), status ),
+                          c_light_gray );
+
+        popup( desc );
+        refresh( recalc, true );
+    }
+    return action;
+}
+
+void comestible_inventory_food::heat_up( item *it_to_heat )
+{
+    //in case we want to warm up a certail quantity later on
+    int batch_size = 1;
+
+    player &p = g->u;
+
+    // basically we want a recipe that satisfies all the requirements for warming food up
+    // water seems exactly what we need, so find it and use it.
+    //might be good idea to create a dummy recipe
+    item water = item( "water_clean", 0 );
+    recipe_subset lr = p.get_learned_recipes();
+    std::vector < const recipe * > water_recipes = lr.search_result( water.typeId() );
+    requirement_data req = water_recipes[0]->requirements();
+
+    const std::vector<std::vector<tool_comp>> &req_tools = req.get_tools();
+    const std::vector<std::vector<quality_requirement>> &req_qual = req.get_qualities();
+
+    // see if we have required tools (e.g. BOIL 1)
+    bool has_qual = true;
+    for( const auto &i : req_qual ) {
+        for( auto &rq : i ) {
+            if( !p.crafting_inventory().has_quality( rq.type, rq.level, batch_size ) ) {
+                has_qual = false;
+                break;
+            }
+        }
+    }
+    if( !has_qual ) {
+        popup( "You don't have necessary tools to heat up your food." );
```

Comment:
Needs to be translatable - see "doc/TRANSLATING.md".

Btw. would be nice to show *what* is actually missing. Doesn't the `requirement` class have some function for creating such a message?

Reply:
Printing missing stuff is complicated. We need to call has_comps to set everything up first; so it's duplicate work. 
Requirements is built around crafting, and will need to be changed if we want to use it in different context. I think it's a good idea, but I don't know code enough for that =\